### DISCUSSION TITLE
Dream Game - Glossary, Wiki, and state machine?

### DISCUSSION THREAD

**Discussion by generrosity** [dialogic-godot/dialogic#1387] [Q&A]
Hello everyone! 

I've been following via youtube and tutorials, and have always been amazed at what has been achieved so far 😢 but afraid to commit too deep.

Context: I have a writeup of a story that I want to write. The introduction is fairly trivial to impliment, but the rest of the story effectively is a knowledge based investigation, and I have a literal wiki-and-a-half of information in a branching narrative that I'd like to tell as a story. I am aware of the usual 'dream game' warning at this point! 😂 This really will be "Game: The Wiki'ing" (and the original story was refered to as "Game: The Spreadsheetening"!)

I have previously been inspired by a [specific GDC talk "Narrative Sorcery"](https://youtu.be/HZft_U4Fc-U) by the Inkle team. Sorry - I'm not sure if you have seen Ink and Inkle and the games like "Heaven's Vault", or thought of the comparison. I couldn't find any references to it here.  The talk touches on two points - using a forward-moving state machine to manage encounters and knowledge (at 17min), and that those states imply previous states - eg if you have cut the wolf's head off you know that there is a wolf in the forest. This is similar to quests but invisible to the player, and is great for putting conversation conditionals around.

I had some questions that I couldn't figure out or are fairly thin on the documentation, sorry.

- Is there a quest / state machine that is inbuilt to Dialogic, or recommended to work well with it? I see there are variable system that you pre-define. What I'm after is 'simple' syntax to do conditional queries, such as the GDC example of a villager met/told you about a wolf/being grateful. (Implimentation I could see how a external godot ENUM or dialogic variables using integers could be done? Or calling to an external dictionary-with-supersedance class?)

- How dynamic are Glossary Cards? I love the idea of information at the player's fingertips, but the information is collected over time. For example - what I'd like to achieve is that intially players won't know who the antagonist is, but will pick up certain quest milestones or individual flags about their name / role / history / etc, potentially in the wrong order, potentially as a progression of logical leaps.   (Implimentation wise, I suppose the question is if its an object I could query and append/replace, or if I could put 'quest' conditional logic directly into the glossary to interprete and parse on the fly?)

- Can Glossary Card data be accessed? I'd like to implimenting a "wiki book" with its Glossary hyperlinks. (Implimentation wise - this might be a specialized 'conversation' with the glossary / hyperlinks? Or extracting the data and hooking click events on keywords??)

Again - sorry I couldn't find the answers elswhere, and happy to take "no" as answers 😅 Thanks for checking this out.

**Comment by ii4y-studios** [comment]
I haven't actually watched that particular talk, so my answwers are gonna be based on general idea and not how they specifically direct them there

* there's not one built in as-is, but Dialogic 2.0 can allow building in additional custom subsystems to use the system integrated ini t as well, and that would be a good possibilty to add as an additional Subsystem either as a custom event or maybe enven built in at top level in the future! as far as the Variable system goes that's already in place, that can be used in logic in Dialogic timelines, as well as from code in Godot. at the moment it's got a section of pre-defined variables, but hopefuly by 2.0 release it will also have Character-specific and Timeline-specific variables as well 
* theGlossary function I also admittedly haven't looked at the 2.0 implemention myself to see where all they hook, but cards are just a set of data that's always able to be referenced, i dont believe there's any like unlock system with them. but it would also be a good idea for a future addon/subsystem, I actually am using something similar to that idea that I wrote separately for one of my games, which I wrote in the Dialogic 1.x days and so its separate but I had been probably going to make it a custom subsystem when I get around to finally upgrading that part
* it should be easily accessible AFIAK, but again I've not looked at the 2.0 implementation so I'm not  sure

**Reply by generrosity** [reply]
Thanks for your insight!  (hopefully described enough so watching vid wasn't required)

I'm definitely looking at Godot 4 / Dialogic 2

**Comment by Jowan-Spooner** [comment]
Thebardsrc alread made some good points, let me add what I know. In 2.x the glossary is a very simple resource. Also each glossary item has an enabled property (true/false). Right now this whole systrm isn't very complex, but I hole it will be expanded upon.
This means accessing the glossary should be pretty easy an I think you could also add custom information to it.

Also, there is no Quest system built in. Depending on the complexity of your game a relatively simple way would be to have an Autoload singleton (e. g. Quest). In 2.x you can access variables of autoloads in conditions (e. g. `if Quest.quest_3 == 4` and I think even callinh functions should work (e. g. `if Quest.is_quest_completed("wolfquest"):`
Maybe that helps you.
